### ISSUE TITLE
Typedefs for Atom services

### ISSUE THREAD

**Issue by nmote** [flowtype/flow-typed#412]
@zertosh wrote up some of the issues [here](https://github.com/flowtype/flow-typed/issues/16#issuecomment-247115391).

Basically, Atom packages talk to each other via service-hub. Packages can mark themselves as consuming or providing a service. Packages that provide a service specify a function that Atom calls. It then takes the returned value and passes it into the functions specified by the consumers.

So, typedefs should be referenced by service name rather than package name.

So, my questions are:

1. Can I add an `atom-service-hub` folder as a sibling to `npm`?
2. How will users install these typedefs? Would you be open to adding logic that looks in the package.json for the properties that Atom uses (`consumedServices` and `providedServices`) and then installs the relevant typedefs?

**Comment by jeffmo** [comment]
> Can I add an atom-service-hub folder as a sibling to npm?

Definitely, although there will be some plumbing that needs to happen to create the first non-npm namespace. Many aspects of the CLI at the moment assume the npm namespace, so we'll just have to break those out a bit.

> How will users install these typedefs? Would you be open to adding logic that looks in the package.json for the properties that Atom uses (consumedServices and providedServices) and then installs the relevant typedefs?

Definitely. I imagine each kind of namespace is going to have different needs in for "auto-installing" -- so we should consider this as we add support for new namespaces.

As for what the explicit, non-automatic install command should be: I'd sort of always imagined this looking something like `flow-typed install atom-service-hub:blah@1.2.3`. (Using just `:` disambiguates from npm's scoped package naming convention which uses `/`)
